src/ota_metadata/legacy2/metadata.py

```diff
@@ -336,98 +356,117 @@ def download_metafiles(
         finally:
             shutil.rmtree(_download_dir, ignore_errors=True)
 
-    def save_fstable(self, dst: StrOrPath, db_fname: str = FSTABLE_DB) -> None:
-        """Dump the file_table to <dst>/<db_fname>"""
-        with closing(self.connect_fstable()) as _fs_conn, closing(
-            sqlite3.connect(Path(dst) / db_fname)
-        ) as _dst_conn:
-            with _dst_conn as conn:
-                _fs_conn.backup(conn)
-
-            with _dst_conn as conn:
-                conn.execute("VACUUM;")
-                # change the journal_mode back to DELETE to make db file on read-only mount work.
-                # see https://www.sqlite.org/wal.html#read_only_databases for more details.
-                conn.execute("PRAGMA journal_mode=DELETE;")
-
-    def prepare_fstable(self) -> None:
-        """Optimize the file_table to be ready for delta generation use."""
-        _orm = FileTableRegularORM(self.connect_fstable)
-        sort_and_replace(
-            _orm,
-            table_name=_orm.orm_table_name,
-            order_by_col="digest",
-        )
-
     # helper methods
 
     def iter_persist_entries(self) -> Generator[str]:
         with open(self._session_dir / self.PERSIST_META_FNAME, "r") as f:
             for line in f:
                 yield line.strip()[1:-1]
 
-    def iter_dir_entries(self, *, batch_size: int) -> Generator[FileTableDirectories]:
+    def iter_dir_entries(self) -> Generator[DirTypedDict]:
         with FileTableDirORM(self.connect_fstable()) as orm:
-            yield from orm.orm_select_all_with_pagination(batch_size=batch_size)
-
-    def iter_non_regular_entries(
-        self, *, batch_size: int
-    ) -> Generator[FileTableNonRegularFiles]:
-        """Yield entries from base file_table which digest presented in OTA image's file_table.
+            _row_factory = typing.cast(Callable[..., DirTypedDict], sqlite3.Row)
+            # fmt: off
+            yield from orm.orm_select_entries(
+                _row_factory=_row_factory,
+                _stmt = gen_sql_stmt(
+                    "SELECT", "path,uid,gid,mode",
+                    "FROM", FT_DIR_TABLE_NAME,
+                    "JOIN", FT_INODE_TABLE_NAME, "USING", "(inode_id)",
+                )
+            )
```

Comment:
Pylance outputs the following warning.
```
Return type of generator function must be compatible with "Generator[FileTableDirectories | Any, Any, Any]"
  "Generator[FileTableDirectories | Any, None, Unknown]" is not assignable to "Generator[DirTypedDict, None, None]"
    Type parameter "_YieldT_co@Generator" is covariant, but "FileTableDirectories | Any" is not a subtype of "DirTypedDict"
      Type "FileTableDirectories | Any" is not assignable to type "DirTypedDict"
        "FileTableDirectories" is not assignable to "DirTypedDict"Pylance[reportReturnType](https://github.com/microsoft/pylance-release/blob/main/docs/diagnostics/reportReturnType.md)
```

Reply:
In L368 I have `_row_factory = typing.cast(Callable[..., DirTypedDict], sqlite3.Row)`, this should have resolved the problem. 
In my VSCode it doesn't complain typing issue here :thinking: 

Updated: Oh please try a `uv sync`, it might resolve the problem.

Reply:
Thank you! resolved.